src/index.js

```diff
@@ -428,16 +410,14 @@ class Node extends EventEmitter {
   _onStopping () {
     series([
       (cb) => {
-        if (this._modules.peerDiscovery) {
-          // stop all discoveries before continuing with shutdown
-          return parallel(
-            this._discovery.map((d) => {
-              return (_cb) => d.stop(() => { _cb() })
-            }),
-            cb
-          )
-        }
-        cb()
+        // stop all discoveries before continuing with shutdown
+        parallel(
+          this._discovery.map((d) => {
+            d.removeListener('peer', this._peerDiscovered)
+            return (_cb) => d.stop(() => { _cb() })
```

Comment:
Can stop() return an error? If so, should we log it here? If not this can just be `return (_cb) => d.stop(_cb)`

Reply:
Yes they can, good call. logs added.